R/create-test-framework.R

```diff
@@ -3,7 +3,7 @@
 #'
 #' @inheritParams create_metworx_docs
 #'
-#' @keywords internal
+#' @export
```

Comment:
I found something weird in here (actually, I originally found it in `join_specs_to_tests()`, but it's in here too.

# Example, using the current main branch of bbr

I see this message (I think from `dplyr`):

<img width="403" alt="Screen Shot 2022-09-01 at 5 31 33 PM" src="https://user-images.githubusercontent.com/7806083/188017484-c3fccfbb-77a2-47aa-88b7-11789779f15f.png">

And then I see the `...2` column in `naw$tests` (looks like maybe just an index?)
<img width="633" alt="Screen Shot 2022-09-01 at 5 32 03 PM" src="https://user-images.githubusercontent.com/7806083/188017563-50acd3d1-18c9-42a5-837d-74f0947fcff2.png">

And, probably more important, I see the `...7` column in `naw$dd`, which I think is supposed to be the count of tests (or maybe tests passed).

<img width="614" alt="Screen Shot 2022-09-01 at 5 32 29 PM" src="https://user-images.githubusercontent.com/7806083/188017650-26f943ba-6ef9-4c91-88ea-313de81948f6.png">


# Disclaimer 

Honestly, this isn't the biggest deal (because it doesn't come through in the finished docs and these are really just exported as debugging helpers) and I'd hate to see you get sidetracked on this when we're so close to release. That said, throw a little time at it and see if you can fix it and, if not, make an issue for it and we'll get it later.

Reply:
It turns out this is due to row names being saved out as an additional column. We could fix this in mrgvalprep or mrgvalidate, but not both. 

At least the first case occurs during `read_csv_test_results`, specifically during the `read_csv` section:
```r
  results <- map(
    csv_files,
    ~{
      read_csv(.x,
               col_types = cols(TestName = "c", TestId = "c",
                                passed = "i", failed = "i")) %>%
        add_column(
          result_file = str_replace(basename(.x), fixed(".csv"), ""),
          .before = TRUE)
    }) %>%
    reduce(rbind)
```

One solution is to use `read.csv(row.names = 1)`, but then we couldnt set the column types the same way (would likely need a mutate call). Alternatively, we could change that code to the following (which I dont like, because tibbles dont have rownames, and they're better for memory storage/viewing)

```r
  results <- map(
    csv_files,
    ~{
      csv_dat_x <- read_csv(.x,
               col_types = cols(TestName = "c", TestId = "c",
                                passed = "i", failed = "i")) %>% as.data.frame()
      rownames(csv_dat_x) <- csv_dat_x[, 1]
      csv_dat_x <- csv_dat_x[, -1]
      csv_dat_x %>%
        add_column(
          result_file = str_replace(basename(.x), fixed(".csv"), ""),
          .before = TRUE)
    }) %>%
    reduce(rbind)
```

Reply:
Ended up going with this change (which fixed all occurrences of that):

```r
  results <- map(
    csv_files,
    ~{
      read.csv(.x, row.names = 1,
               colClasses = c(TestName = "character", TestId = "character",
                                passed = "integer", failed = "integer")) %>%
        add_column(
          result_file = str_replace(basename(.x), fixed(".csv"), ""),
          .before = TRUE)
    }) %>%
    reduce(rbind) %>% as_tibble()
```

Reply:
Ok, I think I follow this. Can you file an issue over [there](https://github.com/metrumresearchgroup/mrgvalprep/issues) with these details and a link to this PR? Thanks.